spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,336 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe ArticlesController, type: :controller do
+  let(:user) { create :user, :role_article_meta }
+  let(:article_category_a) { create :article_category, name: "AAAA" }
+  let(:article_category_b) { create :article_category, name: "BBBB" }
+  let(:article_a) { create :article, name: 'AAAA', note: "AAAA", unit: '250 g', article_category: article_category_a, availability: false }
+  let(:article_b) { create :article, name: 'BBBB', note: "BBBB", unit: '500 g', article_category: article_category_b, availability: true }
+  let(:article_c) { create :article, name: 'CCCC', note: "CCCC", unit: '500 g', article_category: article_category_b, availability: true }
+
+  let(:supplier) { create :supplier, articles: [article_a, article_b] }
+  let(:order) { create :order }
+
+  before { login user }
+
+  describe 'GET index' do
+    it 'assigns sorting on articles' do
+      sortings = [
+        ['name', [article_a, article_b]],
+        ['name_reverse', [article_b, article_a]],
+        ['unit', [article_a, article_b]],
+        ['unit_reverse', [article_b, article_a]],
+        ['article_category', [article_a, article_b]],
+        ['article_category_reverse', [article_b, article_a]],
+        ['note', [article_a, article_b]],
+        ['note_reverse', [article_b, article_a]],
+        ['availability', [article_a, article_b]],
+        ['availability_reverse', [article_b, article_a]]
+      ]
+      sortings.each do |sorting|
+        get :index, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id, sort: sorting[0] }
+        expect(response).to have_http_status(:success)
+        expect(assigns(:articles).to_a).to eq(sorting[1])
+      end
+    end
+
+    it 'triggers an article csv' do
+      get :index, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id }, format: :csv
+      expect(response.header["Content-Type"]).to include("text/csv")
+      expect(response.body).to include(article_a.unit, article_b.unit)
+    end
+  end
+
+  describe "new" do
+    it 'renders form for a new article' do
+      get :new, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id }, xhr: true
```

Comment:
thins applies to all lines of this kind: i think there is a recurring schema withe the `get`. I think that a function (like `get_with_foodcoop_default_scope_and_created_supplier` with a better and shorter name) would help reducing the code duplication and increase readability